loki/expression/expr_visitors.py

```diff
@@ -149,17 +149,10 @@ class FindExpressions(ExpressionFinder):
     (i.e., :class:`pymbolic.primitives.Expression`) in an IR tree.
 
     See :any:`ExpressionFinder`
-
-    Parameters
-    ----------
-    recurse_to_parent : bool, optional
-        For symbols that belong to a derived type, recurse also to the
-        ``parent`` of that symbol (default: `True`)
     """
 
-    def __init__(self, recurse_to_parent=True, **kwargs):
-        self._retriever = ExpressionRetriever(lambda e: isinstance(e, Expression),
-                                              recurse_to_parent=recurse_to_parent)
+    def __init__(self, **kwargs):
+        self._retriever = ExpressionRetriever(lambda e: isinstance(e, Expression))
```

Comment:
This still makes the behaviour object-bound, instead of class-bound and the fact that this is static is applied post-hoc. I wonder it would be easier instead to just set a static `ExpressionRetriever` (expression visitor) via a class method and change the `ExpressionFinder` (IR visitor) to always call `self.retriever.retrieve()`. So for the sub-classes we would then merely do
```
class FindTypedSymbols(ExpressionFinder):

    retriever = ExpressionRetriever(lambda x: ...)
```
without ever touching the `__init__` of the derived instance?